fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -38,12 +39,131 @@
+    """Binary predictor that thresholds continuous predictions of a base estimator.
+
+    At prediction time, the predictor takes as input both standard and sensitive features.
+    Based on the values of sensitive features, it then applies a randomized thresholding
+    transformation according to the provided `interpolation_dict`.
+    
+    :param estimator: base estimator
+    :param dict interpolation_dict: maps sensitive feature values to `Bunch` that describes the
+        interpolation transformation via the following fields:
+        - p0, operation0: with probability p0, operation0 is executed
+        - p1, operation1: with probability p1, operation1 is executed
+        - p_ignore, prediction_constant: two optional fields; if present then the result of
+          the draw of operation0 or operation1 is kept with probability 1 - p_ignore, and gets
+          replaced by prediction_constant with probability p_ignore.
+        The numbers p0 and p1 must be non-negative and add up to 1, operation0 and
+        operation1 must be instances of :class:`ThresholdOperation`, and p_ignore must be
+        between 0 and 1.
+    :param bool prefit: if `True` then the base estimator is not fitted in :meth:`fit`.
+    """
+
+    def __init__(self, estimator, interpolation_dict, prefit=False):
+        self.estimator = estimator
+        self.interpolation_dict = interpolation_dict
+        self.prefit = prefit
+
+    def fit(self, X, y, **kwargs):
+        """Fit the estimator.
+
+        If `prefit` is set to `True` then the base estimator is kept as is.
+        Otherwise it is fitted from the provided arguments.
+        """
+        if self.estimator is None:
+            raise ValueError(ESTIMATOR_ERROR_MESSAGE)
+        
+        if not self.prefit:
+            self.estimator_ = clone(self.estimator).fit(X, y, **kwargs)
+        else:
+            self.estimator_ = self.estimator
+        return self
+
+    def _pmf_predict(self, X, *, sensitive_features):
+        """Probabilistic mass function.
+
+        :param X: Feature matrix
+        :type X: numpy.ndarray or pandas.DataFrame
+        :param sensitive_features: Sensitive features to identify groups by, currently allows
+            only a single column
+        :type sensitive_features: Currently 1D array as numpy.ndarray, list, pandas.DataFrame,
+            or pandas.Series
+        :return: array of tuples with probabilities for predicting 0 or 1, respectively. The sum
+            of the two numbers in each tuple needs to add up to 1.
+        :rtype: numpy.ndarray
+        """
+        check_is_fitted(self)
+        _, _, sensitive_feature_vector = _validate_and_reformat_input(
+            X, y=None, sensitive_features=sensitive_features, expect_y=False,
+            enforce_binary_labels=True)
+        base_predictions = np.array(self.estimator_.predict(X))
+        
+        positive_probs = 0.0*base_predictions
```

Comment:
maybe [`np.zeros_like`](https://numpy.org/doc/stable/reference/generated/numpy.zeros_like.html?highlight=zeros_like#numpy.zeros_like)?

Reply:
I ended up keeping as is, because `base_predictions` is `int`, but `positive_probs` needs to be `float`.